polar2grid/writers/hdf5.py

```diff
@@ -0,0 +1,302 @@
+
+        lon_grp = "{}/longitude".format(parent)
+        lat_grp = "{}/latitude".format(parent)
+
+        lon_dataset = FakeHDF5(fh, fname, lon_grp, compression)
+        lon_dataset.check_variable(lon_data, append)
+        lat_dataset = FakeHDF5(fh, fname, lat_grp, compression)
+        lat_dataset.check_variable(lat_data, append)
+
+        return [lon_data, lat_data], [lon_dataset, lat_dataset]
+
+    @staticmethod
+    def create_variable(filename: str, parent, hdf_subgroup: str, dataset_id: xr.DataArray, **kwargs):
+        """Create a hdf5 subgroup and it's attributes."""
+
+        ds_attrs = dataset_id.attrs
+
+        if hdf_subgroup in parent:
+            LOG.warning("Product %s already in hdf5 group," "will delete existing dataset", hdf_subgroup)
+            del parent[hdf_subgroup]
+
+        try:
+            dset = parent.create_dataset(hdf_subgroup, shape=dataset_id.shape, dtype=dataset_id.dtype, **kwargs)
+        except ValueError:
+            if os.path.isfile(filename):
+                os.remove(filename)
+            raise
+
+        dset.attrs["satellite"] = ds_attrs["platform_name"]
+        dset.attrs["instrument"] = ds_attrs["sensor"]
+        dset.attrs["begin_time"] = ds_attrs["start_time"].isoformat()
+        dset.attrs["end_time"] = ds_attrs["end_time"].isoformat()
+
+        return
+
+    def save_datasets(self, dataset : List[xr.DataArray], filename=None, dtype=None, fill_value=None, append=True, compute=True, **kwargs):
+        """Save hdf5 datasets."""
+        compression = kwargs.pop("compression", None) if "compression" in kwargs else None
+        if compression == "none":
+            compression = None
+
+        add_geolocation = kwargs.pop("add_geolocation") if "add_geolocation" in kwargs else False
+
+        # will this be written to one or multiple files?
+        output_names = []
+        for dataset_id in dataset:
+            args = self._output_file_kwargs(dataset_id)
+            out_filename = filename or self.get_filename(**args)
+            output_names.append(out_filename)
+        one_file = all_equal(output_names)
+
+        filename = output_names[0]
+        if not all_equal(output_names):
+            LOG.warning("Writing to {}".format(filename))
+
+        delayed_write = []
+        hdf5_fh = self.open_hdf5_filehandle(filename, append=append)
+
+        datasets_by_area = self.iter_by_area(dataset)
+        for area, data_arrs in datasets_by_area:
+            # open hdf5 file handle, check if group already exists.
+            parent_group = self.create_proj_group(filename, hdf5_fh, area)
+
+            if add_geolocation:
+                chunks = data_arrs[0].chunks
+                dsets, target_fh = self.write_geolocation(
+                    hdf5_fh, filename, parent_group, area, append, compression, chunks
+                )
+            else:
+                dsets = list()
+                target_fh = list()
```

Comment:
I could have my indentation wrong, but aren't you overwriting the `dsets` and `target_fh` for every for loop? So only one area would ever be possible?

Reply:
Urgh!  That definitely could happen.

Reply:
b021a9 defines a new list for each AREA group.